man/tables.Rd

```diff
@@ -19,9 +19,9 @@ tables(mb=TRUE, order.col="NAME", width=80,
 \details{
 Usually \code{tables()} is executed at the prompt, where \code{parent.frame()} returns \code{.GlobalEnv}. \code{tables()} may also be useful inside functions where \code{parent.frame()} is the local scope of the function; in such a scenario, simply set it to \code{.GlobalEnv} to get the same behaviour as at prompt.
 
-Note that on older versions of \R, \code{object.size} may be slow, so setting \code{mb=FALSE} may speed up execution of \code{tables} significantly.
+`mb = utils::object.size` provides a higher and more accurate estimate of size, but may take longer. Its default `units="b"` is appropriate.
 
```

Comment:
we should document the differences (esp. that char/list columns are skipped, and column attributes are skipped (except factor levels).

it may also make sense to leave mb=object.size as the default and set mb=type_size in our suite. my thinking is that in practice tables may have a large part of memory in string columns; hiding that by default may be confusing.

Reply:
I think if people do use `tables()` they use it mainly to get a list of their data.table objects, number of rows and columns, column names, etc. Like a database has ways just to see the tables that exist.  I think they rarely use the MB column. So the slowness of getting a better MB figure definitely gets in the way of the more common desire just to list the tables.

Reply:
under the `mb` argument further up in `.Rd` I did document the differences and esp. : 

> \item{mb}{ a function which accepts a \code{data.table} and returns its size in bytes. By default, \code{type_size} (same as \code{TRUE}) provides a fast lower bound by excluding the size of character strings in R's global cache (which may be shared) and excluding the size of list column items (which also may be shared). A column \code{"MB"} is included in the output unless \code{FALSE} or \code{NULL}. }

char/list columns are not completely skipped: the size of their pointers is included, but what those pointers point to is excluded. That's what I thought I conveyed quite well in mb argument there, no? The character string part applies to both character columns and the levels of factor columns.

It is missing that column attributes are skipped. That could be added. It does say 'fast lower bound', so perhaps words like "for example, by excluding ..." could be added.

Reply:
> I think they rarely use the MB column.

can't claim to speak for a typical user but personally that column is my main use case for tables()

e.g. when I'm doing exploratory stuff in a sandbox session and need to clear out some temp objects, tables() is a quickly one-liner and then I can run rm(x, tmp2, DT2) or w/e to axe the biggest memory hogs

> That's what I thought I conveyed quite well in mb argument there, no?

yes, conveyed well on re-read.

> perhaps words like "for example, by excluding ..." could be added, rather than trying to document exactly what the function does

yes this works for me, over-documenting would also be a mistake in this case -- fortunes(250): Use the Source, Luke!

> not exporting it at least for now

also agree w keeping it private, especially for now

Reply:
yes even if I'm right about what's most common, there will be users like you who always use MB.  As a utility function, the mb= argument could be getOption("datatable.tables.mb", type_size) then?  Then you could set that option to object.size. Also a better name than `type_size` could be `fast_lower_bound`?  That way `tables(mb=fast_lower_bound,...` reads better in the argument list and .Rd.  Maybe the "MB" column could be named differently depending on the mb function used; e.g. "MB_fast_lower_bound" and "MB_object.size" ?  Although, those are a bit long?

Reply:
Maybe just rename `type_size` as `lower_bound` to be a bit shorter so that `mb=lower_bound` and `"MB_lower_bound"` leaves no room for misleading anyone. (It can be left to inference that the reason for the lower_bound is to be faster.)